File: solutions/371.sum-of-two-integers/sum-of-two-integers.js
/**
 * @param {number} a
 * @param {number} b
 * @return {number}
 */
var getSum = function(a, b) {
    let carry = (a&b) << 1;
    let sum = a ^ b;
    while(carry !== 0){
        const lastSum = sum;
        const lastCarry = carry;
        carry = (lastCarry & lastSum) << 1;
        sum = lastCarry^lastSum;
    }
    return sum;
};

File: solutions/371.sum-of-two-integers/sum-of-two-integers.java
class Solution {
    public int getSum(int a, int b) {
        int carry = (a & b) << 1;
        int res = a ^ b;
        while(carry != 0){
            a = carry;
            b = res;
            carry = (a & b) << 1;
            res = a ^ b;
        }
        return res;
    }
}